files/scripts/supervisor-proc-exit-listener

```diff
@@ -158,56 +158,51 @@ def main(argv):
     critical_group_list, critical_process_list = get_critical_group_and_process_list()
 
     process_under_alerting = {}
-    listener_ready = True
+    # Transition from ACKNOWLEDGED to READY
+    childutils.listener.ready()
 
     while True:
-        # Transition from ACKNOWLEDGED to READY
-        if listener_ready:
+        file_descriptor_list = select.select([sys.stdin], [], [], SELECT_TIMEOUT_SECS)[0]
+        if len(file_descriptor_list) != 0:
+            line = file_descriptor_list[0].readline()
+            headers = childutils.get_headers(line)
+            payload = sys.stdin.read(int(headers['len']))
+
+            # We only care about PROCESS_STATE_EXITED events
+            if headers['eventname'] == 'PROCESS_STATE_EXITED':
+                payload_headers, payload_data = childutils.eventdata(payload + '\n')
+
+                expected = int(payload_headers['expected'])
+                processname = payload_headers['processname']
+                groupname = payload_headers['groupname']
+
+                if ((processname in critical_process_list or groupname in critical_group_list)
+                        and expected == 0):
+                    if container_name != "database":
+                        restart_feature = get_autorestart_state(container_name)
+
+                    if container_name == "database" or restart_feature != "disabled":
+                        MSG_FORMAT_STR = "Process '{}' exited unexpectedly. Terminating supervisor '{}'"
+                        msg = MSG_FORMAT_STR.format(payload_headers['processname'], container_name)
+                        syslog.syslog(syslog.LOG_INFO, msg)
+                        os.kill(os.getppid(), signal.SIGTERM)
+                    else:
+                        process_under_alerting[processname] = time.time()
+
+            # Transition from READY to ACKNOWLEDGED
+            childutils.listener.ok()
+
+            # Transition from ACKNOWLEDGED to READY
             childutils.listener.ready()
```

Comment:
Are we positive this doesn't cause issues to switch to READY, but then do the logging before listening again? I wonder if we should use the flag approach and switch to READY at the top of the loop if we need to.

Reply:
I think this will not cause the issue. But let's change to the flag approach.